### PR TITLE
fix-issue1756: panic in dingTalkRotation function

### PR DESCRIPTION
**Pull request by winfredLIN**
#1756 
direct cause: a panic occurred when accessing a nil pointer while retrieving the workflow.assignee in the dingTalkRotation function.

root cause: the status of the dingtalk instance was not updated when the workflow was canceled. as a result, the dingTalkRotation function, which handled the canceled workflow, filtered by the dingtalk instance whose status was not updated correctly.

solution: update the status of the dingtalk instance to “canceled” in the logic for closing the workflow. this will prevent similar issues from occurring in the future.

### CODE REVIEW

File: sqle/pkg/im/im.go [hunk 1]
```diff
@@ -189,14 +190,52 @@ func CancelApprove(workflowID uint) {
 	s := model.GetStorage()
```
Comment (taolx0) [inline sqle/pkg/im/im.go:190]: CancelApprove 函数不仅仅只有钉钉审批,还有其他的,比如飞书...如果是飞书审批,这里逻辑就错了

**Review by taolx0 — APPROVED** [review]

File: sqle/pkg/im/im.go (hunk 1 above)
Comment (winfredLIN) [reply sqle/pkg/im/im.go:190]: 确实，CancelApprove 应该像im.go中其他函数一样区分不同的im类型，比如飞书和钉钉，根据不同的im类型去取消对应的审批，但是这个issue是解决钉钉轮询panic的问题，我觉得应该另外起一个issue来解决CancelApprove函数中没有根据不同的im类型取消对应的审批的逻辑缺陷。

Comment (taolx0) [reply sqle/pkg/im/im.go:190]: 额...刚好我在处理飞书审批,那就我来改吧

File: sqle/model/configuration.go [hunk 2]
```diff
@@ -497,6 +498,24 @@ func (s *Storage) GetDingTalkInstanceByWorkflowID(workflowId uint) (*DingTalkIns
...
+	var dingTalkInstances []DingTalkInstance
+	err := s.db.Model(&DingTalkInstance{}).Where("workflow_id IN (?)", workflowIds).Find(&dingTalkInstances).Error
+	if err != nil {
+		return nil, err
+	}
+	return dingTalkInstances, nil
+}
+
+// batch update ding_talk_instances'status into canceled by workflow_ids
+func (s *Storage) BatchCancelDingTalkInstance(workflowIds []uint) error {
```
Comment (taolx0) [inline sqle/model/configuration.go:511]: 入参可以改成(workflowIds []uint,status string),批量更新钉钉审批状态,更通用点

**Comment by winfredLIN** [comment]
好的😉，谢谢，你改完之后我再拜读一下

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: sqle/pkg/im/im.go (hunk 1 above)
Comment (winfredLIN) [reply sqle/pkg/im/im.go:190]: 好的，谢谢。

**Pull request merged**